google-cloud-spanner/src/main/resources/com/google/cloud/spanner/connection/PG_ClientSideStatements.json

```diff
@@ -0,0 +1,413 @@
+      "executorName": "ClientSideStatementNoParamExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*(?:commit)(?:\\s+transaction|\\s+work)?(?:\\s+and\\s+no\\s+chain)?\\s*\\z",
+      "method": "statementCommit",
+      "exampleStatements": ["commit", "commit transaction", "commit work", "commit and no chain", "commit transaction and no chain", "commit work and no chain"],
+      "examplePrerequisiteStatements": ["begin transaction"]
+    },
+    {
+      "name": "ROLLBACK [TRANSACTION | WORK] [AND NO CHAIN]",
+      "executorName": "ClientSideStatementNoParamExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*(?:rollback)(?:\\s+transaction|\\s+work)?(?:\\s+and\\s+no\\s+chain)?\\s*\\z",
+      "method": "statementRollback",
+      "exampleStatements": ["rollback", "rollback transaction", "rollback work", "rollback and no chain", "rollback transaction and no chain", "rollback work and no chain"],
+      "examplePrerequisiteStatements": ["begin transaction"]
+    },
+    {
+      "name": "START BATCH DDL",
+      "executorName": "ClientSideStatementNoParamExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*(?:start)(?:\\s+batch)(?:\\s+ddl)\\s*\\z",
+      "method": "statementStartBatchDdl",
+      "exampleStatements": ["start batch ddl"]
+    },
+    {
+      "name": "START BATCH DML",
+      "executorName": "ClientSideStatementNoParamExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*(?:start)(?:\\s+batch)(?:\\s+dml)\\s*\\z",
+      "method": "statementStartBatchDml",
+      "exampleStatements": ["start batch dml"]
+    },
+    {
+      "name": "RUN BATCH",
+      "executorName": "ClientSideStatementNoParamExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*(?:run)(?:\\s+batch)\\s*\\z",
+      "method": "statementRunBatch",
+      "exampleStatements": ["run batch"],
+      "examplePrerequisiteStatements": ["start batch ddl"]
+    },
+    {
+      "name": "ABORT BATCH",
+      "executorName": "ClientSideStatementNoParamExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*(?:abort)(?:\\s+batch)\\s*\\z",
+      "method": "statementAbortBatch",
+      "exampleStatements": ["abort batch"],
+      "examplePrerequisiteStatements": ["start batch ddl"]
+    },
+    {
+      "name": "SET AUTOCOMMIT = TRUE|FALSE",
+      "executorName": "ClientSideStatementSetExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*set\\s+autocommit\\s*(?:=)\\s*(.*)\\z",
+      "method": "statementSetAutocommit",
+      "exampleStatements": ["set autocommit = true", "set autocommit = false"],
+      "setStatement": {
+        "propertyName": "AUTOCOMMIT",
+        "separator": "=",
+        "allowedValues": "(TRUE|FALSE)",
+        "converterName": "ClientSideStatementValueConverters$BooleanConverter"
+      }
+    },
+    {
+      "name": "SET SPANNER.READONLY = TRUE|FALSE",
```

Comment:
I know this was not in our documentation, but I am wondering if we should support `TO` as well (e.g. `SET SPANNER.READONLY TO TRUE`).

WDYT?

Reply:
I think that is a good idea. Human users can work around it, but drivers etc. that generate statements cannot, so we should try to support as many PG variants as possible. Done.